src/uct/ib/ud/base/ud_iface.h

```diff
@@ -215,6 +218,33 @@ static UCS_F_ALWAYS_INLINE void uct_ud_leave(uct_ud_iface_t *iface)
     UCS_ASYNC_UNBLOCK(iface->super.super.worker->async);
 }
 
+static UCS_F_ALWAYS_INLINE int
+uct_ud_iface_filter_dgid(uct_ud_iface_t *iface,void *grh_end,
+                         void *desc, int is_grh_present)
+{
+    void *dgid, *sgid;
+
+    if (!iface->config.filter_enabled) {
+        return 0;
+    }
+
+    if (ucs_unlikely(!is_grh_present)) {
+        ucs_error("RoCE packet does not contain GRH");
+        return 0;
+    }
+
+    sgid = (char*)iface->super.gid.raw + (16 - iface->config.dgid_len);
+    dgid = (char*)grh_end - iface->config.dgid_len;
+
+    if (memcmp(sgid, dgid, iface->config.dgid_len)) {
+        ucs_mpool_put_inline(desc);
```

Comment:
better release desc outside the function. the function should focus on just checking the condition and not cause any side-effects.